rs/https_outcalls/consensus/src/payload_builder.rs

```diff
@@ -282,16 +289,35 @@ impl CanisterHttpPayloadBuilderImpl {
 
             let candidates_and_divergences = response_candidates_by_callback_id
                 .into_iter()
-                .filter_map(|(_, grouped_shares)| {
-                    if let Some((metadata, shares)) = grouped_shares.iter().find(|(_, shares)| {
-                        unique_responses_count += 1;
-                        let signers: BTreeSet<_> =
-                            shares.iter().map(|share| share.signature.signer).collect();
-                        // We need at least threshold different signers to include the response
-                        signers.len() >= threshold
-                    }) {
-                        // A set of grouped shares large enough to meet the
-                        // threshold was found, we should produce a result.
+                .filter_map(|(id, grouped_shares)| {
+                    let consensus_candidate =
+                        grouped_shares.iter().find_map(|(metadata, shares)| {
+                            unique_responses_count += 1;
+                            match canister_http_request_contexts
+                                .get(&id)
+                                .map(|context| &context.replication)
+                            {
+                                Some(Replication::NonReplicated(node_id)) => {
+                                    // For a non-replicated call, we require EXACTLY ONE share,
+                                    // and it MUST be from the designated node.
+                                    shares
+                                        .iter()
+                                        .find(|share| share.signature.signer == *node_id)
+                                        .map(|correct_share| (metadata, vec![*correct_share]))
+                                }
+                                _ => {
```

Comment:
maybe: 
```suggestion
                                None | Some(Replication::Replicated) => {
```
so that if we ever add a new variant to `Replication` the compiler will complain prompting us to decide what to do with the new variant here

Reply:
Good point. changed